.github/workflows/ci.yaml

```diff
@@ -21,8 +21,6 @@ jobs:
   unit-test:
     strategy:
       fail-fast: false
-      matrix:
-        juju-version:  ["2.9", "3.1"]
     name: Unit test charm
```

Comment:
This matrix enables us to test (at least part of the pipelines against Juju 3).

Shouldn't be removed.

Reply:
This is re-running all (44) unit tests with the different values of `LIBJUJU_VERSION_SPECIFIER`, when only 4 of them require special handling of that environment value. So the other 40 are just being re-run with the same configuration. That seems rather wasteful. Or are we missing something here?